src/main/java/net/jodah/failsafe/internal/util/CircularBitSet.java

```diff
@@ -22,8 +22,24 @@ public CircularBitSet(int size, CircularBitSet oldBitSet) {
 
     // Initialize from oldBitSet
     if (oldBitSet != null) {
-      for (int i = 0; i < size && i < oldBitSet.occupiedBits; i++)
+      initializeFromOldBits(oldBitSet);
+    }
+  }
+
+  /**
+   * Initialize bitSet from oldBitSet with correct order
+   */
+  private void initializeFromOldBits(CircularBitSet oldBitSet) {
```

Comment:
Got it! Then I think current behavior on BitSet is on our expectation.
I add the unit test to the PR. Please have a look on it and verify if there is any other issues, thank you!